documentation/src/main/asciidoc/topics/ref_performance_tuning.adoc

```diff
@@ -79,6 +79,78 @@ The recommended values are:
 * send window size - 640 KB
 * receive window size - 25 MB
 
+== Thread pool settings
+
+A thread pool manages threads by assigning them concurrent tasks.
+Additionally, a thread pool can improve system performance by using any idle thread instead of creating threads.
+
+Thread pools limit the creation and deletion of threads to active worker threads, so that the amount of threads does not typically exceed the amount of tasks.
+This helps reduce the memory footprint of your {brandname} server.
+
+Your {brandname} configuration applies default settings to thread pools, but you can change these settings to meet your needs.
+
+The following table outlines common thread pool properties and their default settings:
+
+[cols="1,1"]
+|===
+| Property | Default setting | Description
+
+|`ASYNC_NOTIFICATION_EXECUTOR` |`1`| Used by asynchronous listeners that your {brandname} server registers in embedded mode. 
+The default setting ensures that Infinispan invokes asynchronous listeners according to importance of their messages. 
```

Comment:
Isn't it 'in order of occurence' and not importance?

Reply:
Hi @wfink . I based the order of table entries in alphabetic order. This provides easier scanning and updating for table entries.

Reply:
Think you misunderstood Wolf's point. He is referring to the last few words of this sentence. It should be more like The default setting of one thread ensures that Infinispan invokes asynchronous listeners in the order of occurrence. Probably should mention that if the thread pool size is increased that events could be handled in a different order than they were generated.